sorc/global_cycle.fd/cycle.f90

```diff
@@ -752,25 +776,86 @@ SUBROUTINE SFCDRV(LUGB, IDIM,JDIM,LENSFC,LSOIL,DELTSFC,  &
         SLC_BCK = SLCFCS
 
         ! below updates [STC/SMC/STC]FCS to hold the analysis
-        CALL ADD_INCREMENT_SOIL(RLA,RLO,STCFCS,SMCFCS,SLCFCS,STC_UPDATED, SLC_UPDATED, &
-                LANDINC_MASK,LANDINC_MASK_FG,LENSFC,LSOIL,IDIM,JDIM,LSM,MYRANK)
+        CALL ADD_GSI_INCREMENT_SOIL(RLA,RLO,STCFCS,SMCFCS,SLCFCS,STC_UPDATED, SLC_UPDATED, &
+                STC_INC_TMP,SLC_INC_TMP,LANDINC_MASK,LANDINC_MASK_FG,LENSFC,LSOIL,IDIM,JDIM,LSM,MYRANK)
+
+        ! when read in GSI-based increments, save interpolated fields onto cubed
+        ! sphere tiles
+        CALL WRITE_DATA(LENSFC,IDIM,JDIM,LSOIL,DO_NSST,.true.,NSST, &
```

Comment:
We talked about the issue in the previous round on 01/26/2024 and decided to just leave it for now in case someone will need such a functionality. If you change your mind, please feel free to let me know and I am happy to remove the write out step.